compiler/noirc_evaluator/src/ssa/interpreter/value.rs

```diff
@@ -434,7 +434,34 @@ impl std::fmt::Display for ArrayValue {
         let rc = self.rc.borrow();
 
         let is_slice = if self.is_slice { "&" } else { "" };
-        write!(f, "rc{rc} {is_slice}[")?;
+        write!(f, "rc{rc} {is_slice}")?;
+
+        // Check if the array could be shown as a string literal
+        if self.element_types.len() == 1
+            && matches!(self.element_types[0], Type::Numeric(NumericType::Unsigned { bit_size: 8 }))
+        {
+            const FORM_FEED: u8 = 12; // This is the ASCII code for '\f', which isn't a valid escape sequence in strings
+            let printable = self.elements.borrow().iter().all(|value| {
+                matches!(value, Value::Numeric(NumericValue::U8(byte)) 
+                  if *byte != FORM_FEED && 
```

Comment:
```suggestion
                  *byte != FORM_FEED && 
```
do we need this if?

Reply:
It's the `if` for the `matches!` guard. The logic is also in the evaluator. I thought we couldn't use that but I just pushed a commit to do it.